components/server/src/api/workspace-service-api.ts

```diff
@@ -92,4 +96,16 @@ export class WorkspaceServiceAPI implements ServiceImpl<typeof WorkspaceServiceI
         response.pagination.total = resultTotal;
         return response;
     }
+
+    async createAndStartWorkspace(req: CreateAndStartWorkspaceRequest): Promise<CreateAndStartWorkspaceResponse> {
+        // We can't call WorkspaceService.createWorkspace since there are lot's of parameters checking in GitpodServerImpl.createWorkspace.
+        // TODO: left it as unimplemented until we move logic into WorkspaceService or process source with ContextService
+        throw new ApplicationError(ErrorCodes.INTERNAL_SERVER_ERROR, "not implemented");
```

Comment:
COnnectError(unimplemened)

Reply:
@mustard-mh It makes sense to use ConnectError for `unimplemented`, but I have a question that do we use `ConnectError` for those invalidate argument? I use `ApplicationError` in `listWorkspaces`.

service-api like organization-service-api is using ApplicationError for arg checking too

Reply:
We aligned that we should use ApplicationErrors on application services.

You probably right we shoudl align and use them everywhere not ConnectError 😬 Otherwise it is not consistent.